mailbox/crypto.go

```diff
@@ -120,3 +123,23 @@ func stretchPassword(password []byte) ([]byte, error) {
 
 	return rawPairingBytes, nil
 }
+
+// deriveSID calculates the SID to be used. If the remote key is not present,
+// then only the password will be used to derive the SID. Otherwise, the
+// ECDH operation on the remote and local key will be used.
+func deriveSID(localKey keychain.SingleKeyECDH, remoteKey *btcec.PublicKey,
+	password []byte) ([64]byte, error) {
+
+	var (
+		entropy = password
+		err     error
+	)
+	if remoteKey != nil {
+		entropy, err = ecdh(remoteKey, localKey)
```

Comment:
@Roasbeef do we need to mix in some constant or static value or is it fine to just do `SHA512(shared_key)` (where `shared_key = SHA256(k*P)`)?

Reply:
It can be anything, just something to further bind the secret to this context, and we can use that same secret if we want to generate other random values in a specific context.

Reply:
cool, the current PR includes this 👍